### ISSUE TITLE
Usage service improvements

### ISSUE THREAD

**Issue by robertapplin** [mantidproject/mantid#36311]
**Is your feature request related to a problem? Please describe.**
There are a couple of caveats with regards to the completeness of the usage statistics. In particular, it is possible for users to opt-out of sending usage statistics, so we may not get a full picture of everything which is used. There seems to be two broad themes why people might want to opt-out:
 
1. They want to keep privacy. However, it’s worth noting the usage statistics we collect are completely anonymous and contain no personal data. It’s essentially just an increment to the particular feature that was used. We should then ask, why give the option to opt-out?
 
2. They are concerned reporting usage will cause a slowdown in their workflow (as it sends a usage report each time an algorithm is 
executed). At SNS, they turn off usage statistics by default for this reason. A potential solution would be to only report usage statistics upon exiting Mantid, so usage statistics are only sent once per Mantid session.

**Describe the solution you'd like**
The first thing we should look into is to make sure the sending of usage statistics is done optimally, so that there is no concern that sending usage statistics will cause a slowdown.

We can then decide if we want to still allow users to opt-out of sending usage statistics. This should be discussed at the TWG.

**To summarise**
- [ ] Send usage statistics in an optimal way
- [ ] Check if we can remove the opt-out option for usage statistics
- [ ] Make sure usage statistics are not recorded for Debug builds
- [ ] Create a python script for querying the usage reporter database and returning useful statistics
- [ ] This script can then be used for our algorithm deprecation policy.

**Additional context**
When we have a complete picture, we can use the usage statistics to create an algorithm deprecation policy. The usage statistics could also be useful in Mantid Programme Board meetings.
